### ISSUE TITLE
How do I get VirtualBox to run with linux-xanmod?

### ISSUE THREAD

**Issue by Joe23232** [xanmod/linux#92]
I am using an Arch-based Linux distro. With the `linux` kernel in its vanilla flavour, I am able to run VirtualBox just fine. But with `linux-xanmod` it is unable to run.

When running `virtualbox` through terminal here are the warning/error messages I get:

```
WARNING: The vboxdrv kernel module is not loaded. Either there is no module
         available for the current kernel (5.8.14-xanmod1-1) or it failed to
         load. Please recompile the kernel module and install it by

           sudo /sbin/vboxconfig

         You will not be able to start VMs until this problem is fixed.
Qt WARNING: Icon theme "gnome" not found.
Qt WARNING: Icon theme "ubuntu-mono-dark" not found.
Qt WARNING: Icon theme "Mint-X" not found.
Qt WARNING: Icon theme "elementary" not found.
Qt WARNING: Icon theme "gnome" not found.
```

I have tried running `sudo /sbin/vboxconfig` but here is the error message I get:

```
sudo: /sbin/vboxconfig: command not found
```

Is there a solution to get VirtualBox to run?

**Comment by figue** [comment]
Just install virtualbox-host-dkms package. It will trigger the compilation of VBox's modules in every kernel install/upgrade

**Comment by Joe23232** [comment]
@figue 

Thanks I will give that a try

**Comment by figue** [comment]
@Joe23232 please close the issue if is resolved.

**Comment by MithicSpirit** [comment]
On ubuntu and I have the same issue. Tried reinstalling with `sudo apt-get install --reinstall virtualbox-dkms` but that just led this error:
<details>
<summary> Error log /var/lib/dkms/virtualbox/6.1.10/build/make.log</summary>

```
DKMS make.log for virtualbox-6.1.10 for kernel 5.9.7-xanmod1 (x86_64)
Tue 10 Nov 2020 05:36:16 PM EST
make: Entering directory '/usr/src/linux-headers-5.9.7-xanmod1'
  CC [M]  /var/lib/dkms/virtualbox/6.1.10/build/vboxdrv/linux/SUPDrv-linux.o
  CC [M]  /var/lib/dkms/virtualbox/6.1.10/build/vboxdrv/SUPDrv.o
  CC [M]  /var/lib/dkms/virtualbox/6.1.10/build/vboxdrv/SUPDrvGip.o
  CC [M]  /var/lib/dkms/virtualbox/6.1.10/build/vboxdrv/SUPDrvSem.o
  CC [M]  /var/lib/dkms/virtualbox/6.1.10/build/vboxdrv/SUPDrvTracer.o
  CC [M]  /var/lib/dkms/virtualbox/6.1.10/build/vboxdrv/SUPLibAll.o
  CC [M]  /var/lib/dkms/virtualbox/6.1.10/build/vboxdrv/r0drv/alloc-r0drv.o
  CC [M]  /var/lib/dkms/virtualbox/6.1.10/build/vboxdrv/r0drv/initterm-r0drv.o
  CC [M]  /var/lib/dkms/virtualbox/6.1.10/build/vboxdrv/r0drv/memobj-r0drv.o
  CC [M]  /var/lib/dkms/virtualbox/6.1.10/build/vboxdrv/r0drv/mpnotification-r0drv.o
  CC [M]  /var/lib/dkms/virtualbox/6.1.10/build/vboxdrv/r0drv/powernotification-r0drv.o
  CC [M]  /var/lib/dkms/virtualbox/6.1.10/build/vboxdrv/r0drv/linux/assert-r0drv-linux.o
  CC [M]  /var/lib/dkms/virtualbox/6.1.10/build/vboxdrv/r0drv/linux/alloc-r0drv-linux.o
  CC [M]  /var/lib/dkms/virtualbox/6.1.10/build/vboxdrv/r0drv/linux/initterm-r0drv-linux.o
  CC [M]  /var/lib/dkms/virtualbox/6.1.10/build/vboxdrv/r0drv/linux/memobj-r0drv-linux.o
  CC [M]  /var/lib/dkms/virtualbox/6.1.10/build/vboxdrv/r0drv/linux/memuserkernel-r0drv-linux.o
  CC [M]  /var/lib/dkms/virtualbox/6.1.10/build/vboxdrv/r0drv/linux/mp-r0drv-linux.o
  CC [M]  /var/lib/dkms/virtualbox/6.1.10/build/vboxdrv/r0drv/linux/mpnotification-r0drv-linux.o
In file included from /var/lib/dkms/virtualbox/6.1.10/build/vboxdrv/r0drv/linux/assert-r0drv-linux.c:31:
/var/lib/dkms/virtualbox/6.1.10/build/vboxdrv/r0drv/linux/the-linux-kernel.h:141:11: fatal error: linux/smp_lock.h: No such file or directory
  141 | # include <linux/smp_lock.h>
      |           ^~~~~~~~~~~~~~~~~~
compilation terminated.
make[2]: *** [scripts/Makefile.build:283: /var/lib/dkms/virtualbox/6.1.10/build/vboxdrv/r0drv/linux/assert-r0drv-linux.o] Error 1
make[2]: *** Waiting for unfinished jobs....
In file included from /var/lib/dkms/virtualbox/6.1.10/build/vboxdrv/linux/SUPDrv-linux.c:33:
/var/lib/dkms/virtualbox/6.1.10/build/vboxdrv/r0drv/linux/the-linux-kernel.h:141:11: fatal error: linux/smp_lock.h: No such file or directory
  141 | # include <linux/smp_lock.h>
      |           ^~~~~~~~~~~~~~~~~~
compilation terminated.
make[2]: *** [scripts/Makefile.build:283: /var/lib/dkms/virtualbox/6.1.10/build/vboxdrv/linux/SUPDrv-linux.o] Error 1
In file included from /var/lib/dkms/virtualbox/6.1.10/build/vboxdrv/r0drv/linux/alloc-r0drv-linux.c:31:
/var/lib/dkms/virtualbox/6.1.10/build/vboxdrv/r0drv/linux/the-linux-kernel.h:141:11: fatal error: linux/smp_lock.h: No such file or directory
  141 | # include <linux/smp_lock.h>
      |           ^~~~~~~~~~~~~~~~~~
compilation terminated.
make[2]: *** [scripts/Makefile.build:283: /var/lib/dkms/virtualbox/6.1.10/build/vboxdrv/r0drv/linux/alloc-r0drv-linux.o] Error 1
In file included from /var/lib/dkms/virtualbox/6.1.10/build/vboxdrv/r0drv/linux/initterm-r0drv-linux.c:31:
/var/lib/dkms/virtualbox/6.1.10/build/vboxdrv/r0drv/linux/the-linux-kernel.h:141:11: fatal error: linux/smp_lock.h: No such file or directory
  141 | # include <linux/smp_lock.h>
      |           ^~~~~~~~~~~~~~~~~~
compilation terminated.
make[2]: *** [scripts/Makefile.build:283: /var/lib/dkms/virtualbox/6.1.10/build/vboxdrv/r0drv/linux/initterm-r0drv-linux.o] Error 1
In file included from /var/lib/dkms/virtualbox/6.1.10/build/vboxdrv/r0drv/linux/memobj-r0drv-linux.c:31:
/var/lib/dkms/virtualbox/6.1.10/build/vboxdrv/r0drv/linux/the-linux-kernel.h:141:11: fatal error: linux/smp_lock.h: No such file or directory
  141 | # include <linux/smp_lock.h>
      |           ^~~~~~~~~~~~~~~~~~
compilation terminated.
make[2]: *** [scripts/Makefile.build:283: /var/lib/dkms/virtualbox/6.1.10/build/vboxdrv/r0drv/linux/memobj-r0drv-linux.o] Error 1
In file included from /var/lib/dkms/virtualbox/6.1.10/build/vboxdrv/r0drv/linux/memuserkernel-r0drv-linux.c:31:
/var/lib/dkms/virtualbox/6.1.10/build/vboxdrv/r0drv/linux/the-linux-kernel.h:141:11: fatal error: linux/smp_lock.h: No such file or directory
  141 | # include <linux/smp_lock.h>
      |           ^~~~~~~~~~~~~~~~~~
compilation terminated.
make[2]: *** [scripts/Makefile.build:283: /var/lib/dkms/virtualbox/6.1.10/build/vboxdrv/r0drv/linux/memuserkernel-r0drv-linux.o] Error 1
/var/lib/dkms/virtualbox/6.1.10/build/vboxdrv/SUPDrvTracer.o: warning: objtool: .text+0x7: indirect jump found in RETPOLINE build
/var/lib/dkms/virtualbox/6.1.10/build/vboxdrv/SUPDrvTracer.o: warning: objtool: supdrvTracerProbeFireStub() is missing an ELF size annotation
In file included from /var/lib/dkms/virtualbox/6.1.10/build/vboxdrv/r0drv/linux/mpnotification-r0drv-linux.c:31:
/var/lib/dkms/virtualbox/6.1.10/build/vboxdrv/r0drv/linux/the-linux-kernel.h:141:11: fatal error: linux/smp_lock.h: No such file or directory
  141 | # include <linux/smp_lock.h>
      |           ^~~~~~~~~~~~~~~~~~
compilation terminated.
make[2]: *** [scripts/Makefile.build:283: /var/lib/dkms/virtualbox/6.1.10/build/vboxdrv/r0drv/linux/mpnotification-r0drv-linux.o] Error 1
In file included from /var/lib/dkms/virtualbox/6.1.10/build/vboxdrv/r0drv/linux/mp-r0drv-linux.c:31:
/var/lib/dkms/virtualbox/6.1.10/build/vboxdrv/r0drv/linux/the-linux-kernel.h:141:11: fatal error: linux/smp_lock.h: No such file or directory
  141 | # include <linux/smp_lock.h>
      |           ^~~~~~~~~~~~~~~~~~
compilation terminated.
make[2]: *** [scripts/Makefile.build:283: /var/lib/dkms/virtualbox/6.1.10/build/vboxdrv/r0drv/linux/mp-r0drv-linux.o] Error 1
make[1]: *** [scripts/Makefile.build:500: /var/lib/dkms/virtualbox/6.1.10/build/vboxdrv] Error 2
make: *** [Makefile:1787: /var/lib/dkms/virtualbox/6.1.10/build] Error 2
make: Leaving directory '/usr/src/linux-headers-5.9.7-xanmod1'
```
</details>

I am using the stable release, which should be fine (according to [the website](https://xanmod.org/) the issue only happens on the EDGE and RT kernels).
